### ISSUE TITLE
Bug in children assignment and family expansion

### ISSUE THREAD

**Issue by bgyori** [sorgerlab/indra#168]
The implementation at
https://github.com/sorgerlab/indra/blob/master/indra/preassembler/hierarchy_manager.py#L45-L52
is not perfect, children are sometimes assigned to parents incorrectly.

For instance:
```
from indra.preassembler.hierarchy_manager import hierarchies
hierarchies['entity']._children['http://sorger.med.harvard.edu/indra/entities/mTORC2']

[u'http://identifiers.org/hgnc.symbol/MAPKAP1',
 u'http://identifiers.org/hgnc.symbol/RICTOR',
 u'http://identifiers.org/hgnc.symbol/RPTOR',
 u'http://identifiers.org/hgnc.symbol/MTOR',
 u'http://identifiers.org/hgnc.symbol/MLST8']
```
This is incorrect since RPTOR is not part of mTORC2. It is important to note that both bioentities/relations.csv and indra/resources/entity_hierarchy.rdf are correct in that they don't include RPTOR.

**Comment by bgyori** [comment]
I traced this back to an issue in the transitive closure for `partof` relationships.

**Comment by bgyori** [comment]
So this is pretty interesting... reducing the entity hierarchy to the following lines:
```
<http://identifiers.org/hgnc.symbol/MTOR> <http://sorger.med.harvard.edu/indra/relations/partof> <http://sorger.med.harvard.edu/indra/entities/mTORC1> .
<http://identifiers.org/hgnc.symbol/MTOR> <http://sorger.med.harvard.edu/indra/relations/partof> <http://sorger.med.harvard.edu/indra/entities/mTORC2> .
<http://identifiers.org/hgnc.symbol/RICTOR> <http://sorger.med.harvard.edu/indra/relations/partof> <http://sorger.med.harvard.edu/indra/entities/mTORC2> .
<http://identifiers.org/hgnc.symbol/RPTOR> <http://sorger.med.harvard.edu/indra/relations/partof> <http://sorger.med.harvard.edu/indra/entities/mTORC1> .
```
still results in rdflib telling us that `RPTOR partof+ mTORC2`.

**Comment by bgyori** [comment]
I submitted this: https://github.com/RDFLib/rdflib/issues/715

**Comment by bgyori** [comment]
So this turns out to be an rdflib bug, and to solve this on our end, in the short term, what we can do is replace the `isa+` and `partof+` queries with loops around simple `isa`, `partof` queries.

**Comment by bgyori** [comment]
Fixed in 96d8165a6d26ba1d6ff6ce908daff36713ab2038
